deep_field_metadetect/jaxify/jax_metacal.py

```diff
@@ -0,0 +1,492 @@
+        1 / (1 / obs1.weight + 1 / obs2.weight),
+        0,
+    )
+
+    new_bmask = None
+    new_ormask = None
+    new_noise = None
+    new_mfrac = None
+    new_meta_data = {}
+
+    if obs1.has_bmask() or obs2.has_bmask():
+        new_bmask = _extract_attr(obs1, "bmask", np.int32) | _extract_attr(
+            obs2, "bmask", jnp.int32
+        )
+
+    if obs1.has_ormask() or obs2.has_ormask():
+        new_ormask = _extract_attr(obs1, "ormask", np.int32) | _extract_attr(
+            obs2, "ormask", jnp.int32
+        )
+
+    if obs1.has_noise() or obs2.has_noise():
+        new_noise = _extract_attr(obs1, "noise", np.float32) + _extract_attr(
+            obs2, "noise", jnp.float32
+        )
+
+    if skip_mfrac_for_second:
+        if obs1.has_mfrac():
+            new_mfrac = _extract_attr(obs1, "mfrac", np.float32)
+    else:
+        if obs1.has_mfrac() or obs2.has_mfrac():
+            new_mfrac = (
+                _extract_attr(obs1, "mfrac", np.float32)
+                + _extract_attr(obs2, "mfrac", np.float32)
+            ) / 2  # TODO: update statement
+
+    new_meta_data.update(obs1.meta)
+    new_meta_data.update(obs2.meta)
+
+    obs = NTObservation(
+        image=obs1.image + obs2.image,
+        weight=new_wgt,
+        bmask=new_bmask,
+        ormask=new_ormask,
+        noise=new_noise,
+        jacobian=jax_galsim.wcs.JacobianWCS(
+            dudx=obs1.jacobian.dudx,
+            dudy=obs1.jacobian.dudy,
+            dvdx=obs1.jacobian.dvdx,
+            dvdy=obs1.jacobian.dvdy,
+        ),
+        psf=new_psf,
+        meta=new_meta_data,  # Directly copy metadata
+        mfrac=new_mfrac,
+        store_pixels=getattr(obs1, "store_pixels", True),
+        ignore_zero_weight=getattr(obs1, "ignore_zero_weight", True),
+        jac_row0=obs1.jac_row0,
+        jac_col0=obs1.jac_col0,
+        jac_det=obs1.jac_det,
+        jac_scale=obs1.jac_scale,
+    )
+
+    return obs
+
+
+def get_jax_galsim_object_from_NT_obs(obs, kind="image", rot90=0):
+    """Make an interpolated image from an ngmix obs."""
+    return jax_galsim.InterpolatedImage(
+        jax_galsim.ImageD(
+            jnp.rot90(getattr(obs, kind).copy(), k=rot90),
+            wcs=obs.jacobian,
+        ),
+        x_interpolant="lanczos15",
+    )
+
+
+def get_jax_galsim_object_from_NT_obs_nopix(obs, kind="image"):
+    """Make an interpolated image from an ngmix obs w/o a pixel."""
+    wcs = obs.jacobian
+    return jax_galsim.Convolve(
+        [
+            get_jax_galsim_object_from_NT_obs(obs, kind=kind),
+            jax_galsim.Deconvolve(wcs.toWorld(jax_galsim.Pixel(scale=1))),
+        ]
+    )
+
+
+@partial(
+    jax.jit,
+    static_argnames=[
+        "nxy",
+        "nxy_psf",
+        "reconv_psf_dk",
+        "shears",
+        "skip_obs_wide_corrections",
+        "skip_obs_deep_corrections",
+        "return_noshear_deep",
+    ],
+)
+def jax_helper_metacal_wide_and_deep_psf_matched(
```

Comment:
This function looks like it should be private (i.e., have the name `_jax_helper_metacal_wide_and_deep_psf_matched`) since we do not expect people to use it and instead we expect them to call it via the function below.